mcdc/main.py

```diff
@@ -140,6 +156,37 @@ def run():
     closeout(mcdc)
 
 
+# =============================================================================
+# utilities for handling discrepancies between input and state types
+# =============================================================================
+
+
+def copy_field(dst, src, name):
```

Comment:
This checker may not be necessary. It is OK if there is a discrepancy between input and state types. If there is a discrepancy, the code will break, so it has to be resolved anyway by the dev who adds new items to the code. Resolving the discrepancy may need to be done manually because it can be specific case-by-case. @braxtoncuneo

Reply:
We decided to keep the checker for good measure as it detects things that are conveniently, but silently, handled by Python/Numpy.